### PR TITLE
Nar module cleanups

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by shlevy** [comment]
Still a draft, will mark ready once the module is up to spec

File: hnix-store-core/src/System/Nix/Internal/Nar.hs
Comment (imalsogreg) [inline hnix-store-core/src/System/Nix/Internal/Nar.hs]: Do we know what this comment was referring to? The whole module, Nar effects? Probably not the Nar newtype.

File: hnix-store-core/src/System/Nix/Internal/Nar.hs
Comment (imalsogreg) [inline hnix-store-core/src/System/Nix/Internal/Nar.hs]: Stale comment

**Review by imalsogreg — CHANGES_REQUESTED** [review]
Good reorg. Only nits to change.

**Comment by imalsogreg** [comment]
(meta). Does `Draft` mean, hold off on commenting for now?

**Comment by shlevy** [comment]
Draft means I'm still expecting more to come (in particular, so far I've only touched NarEffects and still need to go through the rest and add haddocks etc.), but still ready for comments on what's come so far.

Also, I'm trying to make to so every commit is in a workable state, so that if someone *wants* to depend on this before I'm done we can merge in and I can continue the rest later.

**Comment by sorki** [comment]
The last commit [e5f1c7d](https://github.com/haskell-nix/hnix-store/pull/41/commits/e5f1c7d9bcf618c3d1351fb006699285402f42d5) was ported to `hnix-store-nar` in #259 (prompted via #252). 

The rest is kind-of obsolete maybe except for the `RawFilePath` patch but the new `filepath` would be a better fit, see also https://github.com/haskell-nix/hnix-store/issues/237

Closing

**Pull request closed without merging**
